driver-core/src/main/com/mongodb/MongoCredential.java

```diff
@@ -543,4 +633,136 @@ public String toString() {
                 + ", mechanismProperties=<hidden>"
                 + '}';
     }
+
+    /**
+     * The context for the {@link OidcRequestCallback#onRequest(OidcRequestContext) OIDC request callback}.
+     */
+    @Evolving
+    public interface OidcRequestContext {
+        /**
+         * @return The OIDC Identity Provider's configuration that can be used to acquire an Access Token.
+         */
+        IdpInfo getIdpInfo();
+
+        /**
+         * @return The timeout that this callback must complete within.
+         */
+        Duration getTimeout();
+    }
+
+    /**
+     * The context for the {@link OidcRefreshCallback#onRefresh(OidcRefreshContext) OIDC refresh callback}.
+     */
+    @Evolving
+    public interface OidcRefreshContext extends OidcRequestContext {
+        /**
+         * @return The OIDC Refresh token supplied by a prior callback invocation.
+         */
+        String getRefreshToken();
+    }
+
+    /**
+     * This callback is invoked when the OIDC-based authenticator requests
+     * tokens from the identity provider.
+     * <p>
+     * It does not have to be thread-safe, unless it is provided to multiple
+     * MongoClients.
+     */
+    public interface OidcRequestCallback {
+        /**
+         * @param context The context.
+         * @return The response produced by an OIDC Identity Provider
+         */
+        IdpResponse onRequest(OidcRequestContext context);
+    }
+
+    /**
+     * This callback is invoked when the OIDC-based authenticator refreshes
+     * tokens from the identity provider. If this callback is not provided,
+     * then refresh operations will not be attempted.
+     * <p>
+     * It does not have to be thread-safe, unless it is provided to multiple
+     * MongoClients.
+     */
+    public interface OidcRefreshCallback {
+        /**
+         * @param context The context.
+         * @return The response produced by an OIDC Identity Provider
+         */
+        IdpResponse onRefresh(OidcRefreshContext context);
+    }
+
+    /**
+     * The OIDC Identity Provider's configuration that can be used to acquire an Access Token.
+     */
+    @Evolving
+    public interface IdpInfo {
+        /**
+         * @return URL which describes the Authorization Server. This identifier is the
+         * iss of provided access tokens, and is viable for RFC8414 metadata
+         * discovery and RFC9207 identification.
+         */
+        String getIssuer();
+
+        /**
+         * @return Unique client ID for this OIDC client.
+         */
+        String getClientId();
+
+        /**
+         * @return Additional scopes to request from Identity Provider. Immutable.
+         */
+        List<String> getRequestScopes();
+    }
+
+    /**
+     * The response produced by an OIDC Identity Provider.
+     */
+    public static final class IdpResponse {
```

Comment:
Check that this class (or its successor) is renamed to `OidcRequestCallback`.

Reply:
Done